scripts/publish.sh

```diff
@@ -21,13 +21,36 @@ IMAGE_SHA_MATCHED="FALSE"
 AWS_FOR_FLUENT_BIT_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_VERSION)
 AWS_FOR_FLUENT_BIT_STABLE_VERSION=$(cat ../AWS_FOR_FLUENT_BIT_STABLE_VERSION)
 
+PUBLISH_LATEST=$(cat ../linux.version | jq -r '.linux.latest')
+echo "Publish Latest? ${PUBLISH_LATEST}"
+
+# Problem: when we push a new version bump the version number in AWS_FOR_FLUENT_BIT_VERSION file changes
+# but that version is not published immediately. Thus, sync tasks normally
+# sync latest version found in DockerHub. But what if we want to release a non-latest version?
+# then sync tasks need to know this. So we use the script to check for an already published version
+# that's not latest
+
+# this code currenly works because DockerHub returns the only last 100 tags and as of March 2023 we only have 64
+# and it should keep working because dockerhub returns the latest tags first
 docker_hub_image_tags=$(curl -s -S 'https://registry.hub.docker.com/v2/repositories/amazon/aws-for-fluent-bit/tags/?page=1&page_size=250' | jq -r '.results[].name')
 tag_array=(`echo ${docker_hub_image_tags}`)
-AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py linux ${tag_array[@]})
+AWS_FOR_FLUENT_BIT_VERSION_DOCKERHUB=$(./get_latest_dockerhub_version.py linux latest ${tag_array[@]})
+
+# If the AWS_FOR_FLUENT_BIT_VERSION is an older version which is already published to dockerhub
+# and latest is set to false in linux.version, then we sync an older non-latest version.
+# otherwise, normal behavior, sync latest version found in dockerhub
```

Comment:
I can't remember now why I didn't just do the work to make it so that the sync tasks sync any latest images that are in docker hub but not in the repo.